### ISSUE TITLE
Global settings priorities discussion

### ISSUE THREAD

**Issue by melissaperreault** [Shopify/dawn#909]
## Context

After prototyping and regrouping we've come to space where we can talk about priorities divide and conquer.
Below I will list what comes to mind, let's combined and divide on multiples issues after and moving forward. Just feels right to have everything at one place to see what we have to do.

### Meeting MVP expectations

1. Remove from the demo store reference (if not done yet) the following settings categories to align with the latest decision
- Sections
- Dividers and lines
2. By removing `Sections` settings, 2 things we need to introduce
- Image with text section, the text part would be styled by the Text box setting moving forward and the media will renamed managed by Media setting
- Bring `Vertical space between sections` setting to `Layout` category (Formerly named `Grid`) ([Preview of the new hierarchy internal file](https://www.figma.com/file/9FOEO0DFKgNDjiqxMakiLB/Design-Settings-Components?node-id=141%3A10944))
3. Split `Popups` and `Drawers` to 2 distinct categories and follow the Figma guidance
4. Discuss if setting combination conditions should be done in a single PR or more granularly  
- Examples:
   - Do not duplicate border for touching borders when horizontal spacing is set to 0
   - Remove the corner radius when horizontal spacing is set to zero or when the component hug another one.

### Follow up tasks
- Buttons review feedback, I can see where ensuring section styles behave as expected will have a greater impact for this week
- Prioritize setting's feedback for each setting categories
- Define specific range of each setting variables
- Content update, this can come in next week once we (UX) have more guidance to give for official settings values and condition rules for all items. I could own and contribute to content update!

**Comment by kmeleta** [comment]
Those priorities seems reasonable to me. Just some random remarks on some of them..

- **Image with text** updates _should_ be a relatively quick win I'd think.

- For **vertical spacing between sections**, moving the settings to layout should be quick enough, but I'd love if we had time to solve this more globally than previously done in prototypes. Remove section-specific margin overrides where possible, make title markup and margins predictable/consistent across sections, etc. I feel a little more time spent to clean this up now could save time/bugs later.

- Not sure if Kai already has thoughts on the **combination conditions** but we definitely need give some thought/explore as to how we make elements "spatially aware". The approach likely may just have to be more granular and specific to each section, but it'd be nice to see if there are any global rules we can establish somehow.

- Before hearing our current priorities, I did start looking at **button/input feedback** to clean up the shadow weirdness and explore how we may include some built in padding to shadows don't overlap with others. I can pause on this for now, but wanted to share that it was in progress.

**Comment by KaichenWang** [comment]
I think before we start any development, there's some documentation housekeeping to complete:

1. Cross-reference the [settings doc](https://docs.google.com/document/d/15GZ4D6J2k2eZ53st6EQUbdi4qpX_RDiRasPVpbfXZ8k/edit#)’s tables with the updates that @melissaperreault outlined. For example the “Vertical space between sections” which is currently under “Descoped - Sections” would need to be moved under “Layout”

2. Cross-reference the finalized doc’s tables with what’s currently on `all-global-settings` branch and identify which settings need to be removed + added

3. Add test scenarios to the document. For each setting group, identify key areas/edge cases to test. I think it will help when we start making the changes to ensure we’re not missing any use cases. For example, for buttons:

![Screen Shot 2021-11-18 at 11 50 03 AM](https://user-images.githubusercontent.com/3710548/142461160-c0f3875e-e286-4d67-8c39-e5af1a443fbd.png)

Once dev and design has signed off on the finalized doc, we can start the implementation knowing that we all aligned on what settings need to be changed and how to test for them

**Comment by melissaperreault** [comment]
### Additional change to consider in our list

- [ ] Full width ignore corner radius. (I think we do already)
- [ ] Text boxes; we are removing the Padding setting, we'll need to revert to the initial section padding

**Comment by melissaperreault** [comment]
> Cross-reference the settings doc’s tables with the updates that @melissaperreault outlined. For example the “Vertical space between sections” which is currently under “Descoped - Sections” would need to be moved under “Layout”

✅ This is done already! 🎉

**Comment by KaichenWang** [comment]
### Developer action items

@kmeleta 
1. Split Popups and Drawers to 2 distinct categories and follow the Figma guidance
2. Fix shadows on buttons adding to the padding between them to stop weird overlaps

@KaichenWang 
1. Review https://github.com/Shopify/dawn/pull/911 and https://github.com/Shopify/dawn/pull/912
2. Update Image with text section, the text part would be styled by the Text box setting moving forward and the media will renamed managed by Media setting

Also, while working on settings, identify and add test cases (sections/elements affected by settings + edge cases) to the [settings doc](https://docs.google.com/document/d/15GZ4D6J2k2eZ53st6EQUbdi4qpX_RDiRasPVpbfXZ8k/edit#)
